### ISSUE TITLE
Add Input-related events/function

### ISSUE THREAD

**Issue by Dragonfly911117** [ntut-open-source-club/practical-tools-for-simple-design#81]
> 誒話說我寫上面那個才發現 你們Input Class那邊只給了檢測當前frames按鈕是否按下去的function
-- 黑喵

[source](https://discord.com/channels/1093814266630512681/1093938857772322886/1196396173594996806)

- [x] [Key Up Event](https://wiki.libsdl.org/SDL2/SDL_KeyboardEvent)

Feel free to add tasks.

**Comment by NOOBDY** [comment]
fixed in #123

**Comment by ntut-rick** [comment]
[discord](https://ptb.discord.com/channels/1093814266630512681/1207972838590386187)

reference from [LGF](https://github.com/ntut-xuan/LeistungsstarkesGameFramework/blob/3a433d0262f778ace36fe85694876de3833f6ea6/Source/Game/mygame.h#L76-L92):

```cpp
class CGameStateRun : public CGameState {
public:
        CGameStateRun(CGame *g);
        ~CGameStateRun();
        void OnBeginState();                                                    // 設定每次重玩所需的變數
        void OnInit();                                                                  // 遊戲的初值及圖形設定
        void OnKeyDown(UINT, UINT, UINT);
        void OnKeyUp(UINT, UINT, UINT);
        void OnLButtonDown(UINT nFlags, CPoint point);  // 處理滑鼠的動作
        void OnLButtonUp(UINT nFlags, CPoint point);    // 處理滑鼠的動作
        void OnMouseMove(UINT nFlags, CPoint point);    // 處理滑鼠的動作 
        void OnRButtonDown(UINT nFlags, CPoint point);  // 處理滑鼠的動作
        void OnRButtonUp(UINT nFlags, CPoint point);    // 處理滑鼠的動作
protected:
        void OnMove();                                                                  // 移動遊戲元素
        void OnShow();                                                                  // 顯示這個狀態的遊戲畫面
};
 ```
